### PR TITLE
feat: Use util function for hostname building

### PR DESCRIPTION
**Pull request by samuel-hawker**
Signed-off-by: Samuel Hawker <samuel.hawker@ibm.com>

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java [hunk 1]
```diff
@@ -136,7 +136,12 @@ public KafkaBrokerConfigurationBuilder withListeners(String clusterName, String
 
         // Replication listener
         listeners.add("REPLICATION-9091://0.0.0.0:9091");
-        advertisedListeners.add(String.format("REPLICATION-9091://%s-${STRIMZI_BROKER_ID}.%s-brokers.%s.svc:9091", KafkaResources.kafkaStatefulSetName(clusterName), KafkaResources.kafkaStatefulSetName(clusterName), namespace));
+        advertisedListeners.add(String.format("REPLICATION-9091://%s:9091",
+                ModelUtils.podDnsName(namespace,
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java:140]: Should this be `podDnsNameWithoutClusterDomain`?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java [hunk 2]
```diff
@@ -273,7 +278,13 @@ private void configureReplicationListener() {
      * @return  String with advertised listener configuration
      */
     private String getAdvertisedListener(String clusterName, String namespace, String listenerName, int port)    {
-        return String.format("%s://%s-${STRIMZI_BROKER_ID}.%s-brokers.%s.svc:%d", listenerName, KafkaResources.kafkaStatefulSetName(clusterName), KafkaResources.kafkaStatefulSetName(clusterName), namespace, port);
+        return String.format("%s://%s:%d",
+                listenerName,
+                ModelUtils.podDnsName(namespace,
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java:283]: Same as above - should this be `podDnsNameWithoutClusterDomain`?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java [hunk 3]
```diff
@@ -1478,10 +1478,10 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
                             return Future.failedFuture(Util.missingSecretException(namespace, ClusterOperator.secretName(name)));
                         }
 
-                        Function<Integer, String> zkNodeAddress = (Integer i) -> String.format("%s.%s.%s.svc",
-                                zkCluster.getPodName(i),
+                        Function<Integer, String> zkNodeAddress = (Integer i) -> ModelUtils.podDnsName(
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java:1481]: Same as above - should this be `podDnsNameWithoutClusterDomain`?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java (hunk 1 above)
Comment (samuel-hawker) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java:140]: You're absolutely right, I made these changes prior to our conversation and misunderstood. whether they should be long or short.

**Pull request merged**